File: src/main/java/com/demonwav/mcdev/inspection/PlatformAnnotationEntryPoint.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 *
 * MIT License
 */
package com.demonwav.mcdev.inspection;

import com.google.common.collect.ImmutableList;
import com.intellij.codeInspection.reference.EntryPoint;
import com.intellij.codeInspection.reference.RefElement;
import com.intellij.openapi.util.InvalidDataException;
import com.intellij.openapi.util.WriteExternalException;
import com.intellij.psi.PsiElement;
import org.jdom.Element;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class PlatformAnnotationEntryPoint extends EntryPoint {
    @NotNull
    @Override
    public String getDisplayName() {
        return "Minecraft Entry Point";
    }

    @Override
    public boolean isEntryPoint(@NotNull RefElement refElement, @NotNull PsiElement psiElement) {
        return false;
    }

    @Override
    public boolean isEntryPoint(@NotNull PsiElement psiElement) {
        return false;
    }

    @Override
    public boolean isSelected() {
        return false;
    }

    @Override
    public void setSelected(boolean selected) {

    }

    @Nullable
    @Override
    public String[] getIgnoreAnnotations() {
        return ImmutableList.of("org.spongepowered.api.event.Listener", "org.bukkit.event.EventHandler").toArray(new String[2]);
    }

    @Override
    public void readExternal(Element element) throws InvalidDataException {

    }

    @Override
    public void writeExternal(Element element) throws WriteExternalException {

    }
}

File: src/main/kotlin/com/demonwav/mcdev/inspection/PlatformAnnotationEntryPoint.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.inspection

import com.intellij.codeInspection.reference.EntryPoint
import com.intellij.codeInspection.reference.RefElement
import com.intellij.openapi.util.InvalidDataException
import com.intellij.openapi.util.WriteExternalException
import com.intellij.psi.PsiElement
import org.jdom.Element

class PlatformAnnotationEntryPoint : EntryPoint() {
    override fun getDisplayName() = "Minecraft Entry Point"
    override fun isEntryPoint(refElement: RefElement, psiElement: PsiElement) = false
    override fun isEntryPoint(psiElement: PsiElement) = false
    override fun isSelected() = false
    override fun setSelected(selected: Boolean) {}
    override fun getIgnoreAnnotations() =
        arrayOf("org.spongepowered.api.event.Listener", "org.bukkit.event.EventHandler")

    @Throws(InvalidDataException::class)
    override fun readExternal(element: Element) {
    }

    @Throws(WriteExternalException::class)
    override fun writeExternal(element: Element) {
    }
}
